src/auth_token.rs

```diff
@@ -94,20 +94,21 @@ impl AuthToken {
 
     /// Pass a function to check against the token. If a caveat to be checked doesn't exist,
     /// verification fails and 'false' is returned.
-    pub fn verify_caveat(
-        &self,
-        caveat: &str,
-        checker: fn(CaveatContents) -> bool,
-    ) -> NdResult<bool> {
+    pub fn verify_caveat(&self, caveat: &str, checker: fn(CaveatContents) -> bool) -> NdResult<()> {
         // Check caveat w/ name exists....
         let (_caveat_name, caveat_contents) = match &self.get_caveat_by_name(caveat) {
             Some(target_caveat) => target_caveat,
-            None => return Ok(false), // Or should we error here?
+            None => return Err(Error::AccessDenied("No such Caveat".to_string())), // Or should we error here?
```

Comment:
if we're erroring, lets remove the comment at the end there 👍

Reply:
I'm also not 100% on this being an error.

If it doesn't exist, that's a fail. That makes sense and is pretty simple.

If we're throwing errors all we're doing is forcing that logic onto the consumer.... Thoughts @Yoga07 ?

Reply:
We could perhaps log here to make it debuggable...?

Reply:
>If we're throwing errors all we're doing is forcing that logic onto the consumer

Yup, true, but we want the consumer to know _why exactly_ it failed as well(by returning strings). And also, I think it's not a good practice to Result-ify a `bool`. If it's clear we `Ok(())`, if not we state what's wrong via `AccessDenied(String)`. It can handle multiple fail cases hence so.

>We could perhaps log here to make it debuggable...?

Debugs should be done at clients and vaults as we only log them and not safe-nd.

Reply:
Alright. Sold. 👍